src/schemathesis/cli/output/junitxml.py

```diff
@@ -0,0 +1,46 @@
+import platform
+from typing import Optional
+
+import click
+from junit_xml import TestSuite, TestCase, to_xml_report_string
+
+from ..handlers import EventHandler, ExecutionContext
+from . import default
+from ...models import Status
+from ...runner import events
+
+
+class JunitXMLHandler(EventHandler):
+    test_cases = []
+    start_time = None
+
+    def handle_event(self, context: ExecutionContext, event: events.ExecutionEvent) -> None:
+        if isinstance(event, events.Initialized):
+            self.start_time = event.start_time
+        if isinstance(event, events.BeforeExecution):
+            self.test_cases.append(TestCase(f"{event.method} {event.path}",))
+        if isinstance(event, events.AfterExecution):
+            self.test_cases[-1].elapsed_sec = event.elapsed_time
+            if event.status == Status.failure:
+                checks = default._get_unique_failures(event.result.checks)
+                for idx, check in enumerate(checks, 1):
+                    message: Optional[str]
+                    if check.message:
+                        message = f"{idx}. {check.message}"
+                    else:
+                        message = None
+                self.test_cases[-1].add_failure_info(message=message)
+            if event.status == Status.error:
+                self.test_cases[-1].add_error_info(message=event.result.errors[-1].exception)
+        if isinstance(event, events.Finished):
+            ts = [TestSuite("schemathesis", test_cases=self.test_cases, hostname=platform.node(),)]
+            click.echo("------------XML-----------")
+            click.echo(to_xml_report_string(ts, prettyprint=True))
+            status_code = 0
+            if event.has_errors or event.has_failures:
+                status_code = 1
+            raise click.exceptions.Exit(status_code)
```

Comment:
It should not be handled here. it is already handled in the stdout handlers (and will be moved out from there in the future)